### ISSUE TITLE
Add a `Software::Version` class similar to `Gem::Version`

### ISSUE THREAD

**Issue by postmodern** [ronin-rb/ronin-support#510]
Add a `Software::Version` class similar to `Gem::Version`, but accepts messier non-rubygem style versions. It must support parsing and comparing versions.

```
1234
1.2
1.2.3
1.2.3.4
1.2.3-4
1.2.3-4.5
1.2.3-foo
1.2.3.foo
1.2.3_4
1.2.3_4.5
1.2.3_foo
1.2.3+4
1.2.3+4.5
1.2.3+foo
```

This class can be used in [ronin-exploits](https://github.com/ronin-rb/ronin-exploits#readme) to aide in parsing and comparing version numbers.

**Comment by postmodern** [comment]
[SemVer](https://semver.org/#spec-item-10) states that the `+` symbol is used to denote non-sortable build metadata, such as SHA1 hashes. We should ignore everything after a `+` symbol.

**Comment by postmodern** [comment]
Implemented by 1cc2b23.
